frontends/concrete-python/tests/tfhers-utils/src/main.rs

```diff
@@ -48,25 +48,55 @@ fn set_server_key_from_file(path: &String) {
     set_server_key(sk);
 }
 
-fn encrypt_with_key(value: u8, client_key: ClientKey, path: &String) {
+fn encrypt_with_key_u8(
+    value: u8,
+    client_key: ClientKey,
+    ciphertext_path: &String,
+) -> Result<(), Box<dyn std::error::Error>> {
     let ct = FheUint8::encrypt(value, &client_key);
-    safe_save(path, &ct)
+
+    safe_save(ciphertext_path, &ct);
+
+    Ok(())
+}
+
+fn encrypt_with_key_i8(
+    value: i8,
+    client_key: ClientKey,
+    ciphertext_path: &String,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let ct = FheInt8::encrypt(value, &client_key);
+
+    safe_save(ciphertext_path, &ct);
+
+    Ok(())
 }
 
 fn decrypt_with_key(
     client_key: ClientKey,
     ciphertext_path: &String,
     plaintext_path: Option<&String>,
-) {
-    let fheuint: FheUint8 = safe_load(ciphertext_path);
-    let result: u8 = fheuint.decrypt(&client_key);
+    signed: bool,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let string_result: String;
+
+    if signed {
+        let fheint: FheInt8 = safe_load(ciphertext_path);
+        let result: i8 = fheint.decrypt(&client_key);
+        string_result = result.to_string();
+    } else {
+        let fheuint: FheUint8 = safe_load(ciphertext_path);
+        let result: u8 = fheuint.decrypt(&client_key);
+        string_result = result.to_string();
+    }
 
     if let Some(path) = plaintext_path {
         let pt_path: &Path = Path::new(path);
-        fs::write(pt_path, result.to_string()).unwrap();
+        fs::write(pt_path, string_result)?;
     } else {
-        println!("result: {}", result);
+        println!("result: {}", string_result);
     }
+    Ok(())
```

Comment:
Sam here if we always return `Ok()` doesn't make sense. So if we want to correctly handle error, we need to propagate error also from safe_load.

Reply:
related to above